api/v1alpha1/stage_types.go

```diff
@@ -118,6 +118,14 @@ type ImageSubscription struct {
 	//+kubebuilder:validation:MinLength=1
 	//+kubebuilder:validation:Pattern=`^(([\w\d\.-]+)(:[\d]+)?/)?[a-z0-9-]+(/[a-z0-9-]+)*$`
 	RepoURL string `json:"repoURL"`
+	// GitRepoURL optionally specifies the URL of a Git repository that contains
+	// the source code for the image repository referenced by the RepoURL field.
+	// When this is specified, Kargo MAY be able to infer and link to the exact
+	// revision of that source code that was used to build the image.
+	//
+	//+kubebuilder:validation:Optional
+	//+kubebuilder:validation:Pattern=`^(((https?://)|([\w-]+@))([\w\d\.]+)(:[\d]+)?/(.*))?$`
+	GitRepoURL string `json:"gitRepoURL"`
```

Comment:
1.  I feel this should have `,omitempty`. 
2. I'd like to avoid confusion between `repoURL` and `gitRepoURL`. `git` is not actually the differentiator here (e.g. RepoURL might also be git) but rather, it's the URL of the app source code. So I propose instead:
* `appRepoURL`
* `appSourceRepoURL`

Reply:
Let me double-check on `omitempty`. There are some peculiar/counter-intuitive rules about how that impacts the CRD generation.

The second point is a little off, however. "git" _is_ the differentiator here. The struct that this is enclosed in is `ImageSubscription` and `RepoURL` is the URL of an _image_.

I think `GitRepoURL` accurately denotes that this is the source repo that corresponds to that image.

Reply:
Ah... yes. Missing `omitempty` here was an oversight. Thanks for catching it.

What I was thinking of was the reverse -- _having_ omitempty forced a required field to be optional in the CRD, no matter that the kubebuilder annotations say.

Fixing.

Reply:
Fixed in d061a27a28a6230d3f12e755a651bc66ee0e8219

Reply:
> RepoURL is the URL of an image.

Oh, I see. In that case gitRepoURL could work. 

> The struct that this is enclosed in is `ImageSubscription`

If this is only in the `ImageSubscription`, will we be able to show linkage from either (1) a Stage that subscribes to another Stage or (2) a Stage that subscribes to a git location?

Reply:
Hmmmm... not easily. Thanks for pointing that out. Let me think on this. I've got an idea that could work and it would probably lessen the effort on the front end pretty significantly. Going to convert this to a draft for now.

Reply:
@jesse

> If this is only in the ImageSubscription, will we be able to show linkage from either (1) a Stage that subscribes to another Stage or (2) a Stage that subscribes to a git location?

I've updated the PR to account for this. It's in the state now, too.